lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -90,61 +101,82 @@ def import_file(self, params):
+                    self._parse_exon(feature)
+                else:
+                    self._transform_feature(contig, feature)
 
-        aUtil = AssemblyUtil(callback_url)
-        assembly_ref = aUtil.save_assembly_from_fasta(
-                                                {'file':
-                                                    {'path': input_fasta_file,
-                                                     'assembly_name': assembly['assembly_id']},
-                                                 'workspace_name': workspace_name,
-                                                 'assembly_name': assembly['assembly_id']})
+        # save assembly file
+        assembly_ref = self.au.save_assembly_from_fasta(
+            {'file': {'path': input_fasta_file},
+             'workspace_name': workspace_name,
+             'assembly_name': core_genome_name+".assembly"})
+        assembly_data = self.dfu.get_objects(
+            {'object_refs': [assembly_ref],
+             'ignore_errors': 0})['data'][0]['data']
 
         # generate genome info
-        genome = self._gen_genome_info(core_genome_name, scientific_name, assembly_ref,
-                                       genome_features_list, genome_cdss_list, genome_mrnas_list,
-                                       source, assembly, taxon_reference, taxonomy, input_gff_file)
-
-        workspace_id = self.dfu.ws_name_to_id(workspace_name)
-        genome_info = self.dfu.save_objects({"id": workspace_id,
-                                             "objects": [{"name": core_genome_name,
-                                                          "type": "KBaseGenomes.Genome",
-                                                          "data": genome}]})[0]
+        genome = self._gen_genome_info(core_genome_name, scientific_name,
+                                       assembly_ref, source, assembly_data,
+                                       input_gff_file, molecule_type)
+
+        result = self.gi.save_one_genome({
+            'workspace': workspace_name,
+            'name': core_genome_name,
+            'data': genome,
+            "meta": metadata,
+        })
         report_string = ''
 
-        return {'genome_info': genome_info, 'report_string': report_string}
+        return {'genome_info': result['info'], 'report_string': report_string}
 
-    def _validate_import_file_params(self, params):
+    def _parse_exon(self, feature):
+        parent_id = feature.get('Parent', '')
+        if not parent_id:
+            raise ValueError("An exon has no valid parent: {}".format(feature))
+        parent = self.feature_dict[parent_id]
+        if parent.get('exons', False):
+            self.feature_dict[parent_id]['location'].append(
+                self._location(feature))
+        else:
+            parent['exons'] = True
+            self.feature_dict[parent_id]['location'] = [self._location(feature)]
+
+    @staticmethod
+    def _location(in_feature):
```

Comment:
Insure that the contig exists in the assembly and that the location is within the length of the contig.

Reply:
now testing this on line 124 for contig_id and 600 for location

Reply:
Those line numbers do not correspond to what you say they are doing.